### PR TITLE
Feature/dialect support comment

### PR DESCRIPTION
**Pull request by cq-z**
Hello!

* Type: bug fix | new feature | code quality | documentation
* Link to issue:

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Small description of change:

Thanks

### CODE REVIEW

**Comment by ruudboon** [comment]
Linked: https://github.com/phalcon/cphalcon/pull/15085

File: phalcon/Db/Adapter/Pdo/Mysql.zep [hunk 1]
```diff
@@ -401,39 +401,44 @@ class Mysql extends PdoAdapter
...
+                if memstr(field[6], "on update") {
+                    let definition["default"] = "NULL " . field[6];
                 }
             }
-
+            /**
+             * Check if the column has comment
+             */
+             if field[8] !== null {
+                let definition["comment"] = "" . field[8];
```
Comment (Jeckerson) [inline phalcon/Db/Adapter/Pdo/Mysql.zep:440]:
```suggestion
                let definition["comment"] = field[8];
```

File: phalcon/Db/Adapter/Pdo/Postgresql.zep [hunk 2]
```diff
@@ -512,7 +512,12 @@ class Postgresql extends PdoAdapter
                     let definition["default"] = null;
                 }
             }
-
+            /**
+             * Check if the column has comment
+             */
+             if field[10] !== null {
+                let definition["comment"] = "" . field[10];
```
Comment (Jeckerson) [inline phalcon/Db/Adapter/Pdo/Postgresql.zep:519]:
```suggestion
                let definition["comment"] = field[10];
```

File: phalcon/Db/Column.zep [hunk 3]
```diff
@@ -248,6 +249,13 @@ class Column implements ColumnInterface
      */
     protected name { get };
 
+    /**
+     * Column's comment
+     *
+     * @var string
+     */
+     protected comment = null  { get };
```
Comment (Jeckerson) [inline phalcon/Db/Column.zep:257]:
```suggestion
     protected comment = null { get };
```

File: phalcon/Db/Dialect/Mysql.zep [hunk 4]
```diff
@@ -211,6 +211,12 @@ class Mysql extends Dialect
             if column->isPrimary() {
                 let columnLine .= " PRIMARY KEY";
             }
+            /**
```
Comment (Jeckerson) [inline phalcon/Db/Dialect/Mysql.zep:214]:
```suggestion

            /**
```

File: phalcon/Db/Dialect/Mysql.zep [hunk 5]
```diff
@@ -211,6 +211,12 @@ class Mysql extends Dialect
...
+             * Add a COMMENT clause
+             */
+             if column->getComment() {
+                let columnLine .= " COMMENT '".column->getComment()."'";
```
Comment (Jeckerson) [inline phalcon/Db/Dialect/Mysql.zep:218]:
```suggestion
                let columnLine .= " COMMENT '" . column->getComment() . "'";
```

File: phalcon/Db/Dialect/Mysql.zep [hunk 6]
```diff
@@ -732,6 +738,13 @@ class Mysql extends Dialect
             let sql .= " AUTO_INCREMENT";
         }
 
+        /**
+        * Add a COMMENT clause
+        */
+        if column->getComment() {
+            let sql .= " COMMENT '".column->getComment()."'";
```
Comment (Jeckerson) [inline phalcon/Db/Dialect/Mysql.zep:745]:
```suggestion
            let sql .= " COMMENT '" . column->getComment() . "'";
```

File: phalcon/Db/Dialect/Mysql.zep [hunk 7]
```diff
@@ -863,6 +876,15 @@ class Mysql extends Dialect
                     tableOptions[] = "COLLATE=" . tableCollation;
             }
         }
+        /**
```
Comment (Jeckerson) [inline phalcon/Db/Dialect/Mysql.zep:879]:
```suggestion

        /**
```

File: phalcon/Db/Dialect/Mysql.zep [hunk 8]
```diff
@@ -863,6 +876,15 @@ class Mysql extends Dialect
...
+         * Check if there is a TABLE_COMMENT option
+         */
+         if fetch tableComment, options["TABLE_COMMENT"] {
+            if tableComment {
+                let tableOptions = "COMMENT='" . tableComment . "'";
```
Comment (Jeckerson) [inline phalcon/Db/Dialect/Mysql.zep:884]:
```suggestion
                let tableOptions = "COMMENT = '" . tableComment . "'";
```

File: phalcon/Db/Dialect/Mysql.zep [hunk 9]
```diff
@@ -899,4 +921,19 @@ class Mysql extends Dialect
...
+     */
+     public function alterTableComment(string! tableName, string! schemaName, string! comment) -> string
+     {
+         string table;
+ 
+         if schemaName {
+             let table = "`" . schemaName . "`.`" . tableName . "`";
+         } else {
+             let table = "`" . tableName . "`";
+         }
```
Comment (Jeckerson) [inline phalcon/Db/Dialect/Mysql.zep:936]:
```suggestion
         }

```

File: phalcon/Db/Dialect/Postgresql.zep [hunk 10]
```diff
@@ -181,16 +186,20 @@ class Postgresql extends Dialect
             }
 
             let createLines[] = columnLine;
+
+            /**
+            * Add a COMMENT clause
+            */
+            if column->getComment() {
+                let indexSqlAfterCreate .= " COMMENT ON COLUMN " . table . ".\"" . column->getName()."\" is '".column->getComment()."';";
```
Comment (Jeckerson) [inline phalcon/Db/Dialect/Postgresql.zep:194]:
```suggestion
                let indexSqlAfterCreate .= " COMMENT ON COLUMN " . table . ".\"" . column->getName()."\" IS '".column->getComment()."';";
```

File: phalcon/Db/Dialect/Postgresql.zep [hunk 11]
```diff
@@ -250,6 +259,14 @@ class Postgresql extends Dialect
         let sql .= join(",\n\t", createLines) . "\n)";
         if isset definition["options"] {
             let sql .= " " . this->getTableOptions(definition);
+            /**
+            * Check if there is a TABLE_COMMENT option
+            */
+            if fetch tableComment, definition["options"]["TABLE_COMMENT"] {
+                if tableComment {
+                    let indexSqlAfterCreate = " COMMENT ON TABLE " . table . " IS 'Our session logs';";
```
Comment (Jeckerson) [inline phalcon/Db/Dialect/Postgresql.zep:267]: Why here is hardcoded comment?

File: phalcon/Db/Dialect/Postgresql.zep [hunk 12]
```diff
@@ -583,6 +600,13 @@ class Postgresql extends Dialect
             }
         }
 
+        /**
+         * Add a COMMENT clause
+         */
+         if column->getComment() {
+            let sql .= "COMMENT ON COLUMN " . this->prepareTable(tableName, schemaName) . ".\"" .column->getName()."\" is '".column->getComment()."';";
```
Comment (Jeckerson) [inline phalcon/Db/Dialect/Postgresql.zep:607]:
```suggestion
            let sql .= "COMMENT ON COLUMN " . this->prepareTable(tableName, schemaName) . ".\"" .column->getName()."\" IS '" . column->getComment() . "';";
```

File: phalcon/Db/Dialect/Postgresql.zep [hunk 13]
```diff
@@ -700,4 +728,19 @@ class Postgresql extends Dialect
...
+     public function alterTableComment(string! tableName, string! schemaName, string! comment) -> string
+     {
+         string table;
+ 
+         if schemaName {
+             let table = "`" . schemaName . "`.`" . tableName . "`";
+         } else {
+             let table = "`" . tableName . "`";
+         }
+         return "COMMENT ON TABLE " . table . " IS '" . comment . "'";
```
Comment (Jeckerson) [inline phalcon/Db/Dialect/Postgresql.zep:744]:
```suggestion

         return "COMMENT ON TABLE " . table . " IS '" . comment . "'";
```

File: tests/tocheck-database/Db/Dialect/Postgresql/CreateTableCest.php [hunk 14]
```diff
@@ -213,6 +213,51 @@ protected function getCreateTableFixtures(): array
...
+                            'autoIncrement' => true,
+                            'first'         => true,
+                            'comment'       => 'test',
+                        ]),
+                        new Column('column15', [
+                            'type'    => Column::TYPE_INTEGER,
+                            'default' => 5,
+                            'notNull' => true,
+                            'after'   => 'user_id',
+                            'comment'       => 'test',
```
Comment (Jeckerson) [inline tests/tocheck-database/Db/Dialect/Postgresql/CreateTableCest.php:232]:
```suggestion
                            'comment' => 'test',
```

File: tests/tocheck-database/Db/Dialect/Postgresql/CreateTableCest.php [hunk 15]
```diff
@@ -213,6 +213,51 @@ protected function getCreateTableFixtures(): array
...
+                        new Column('column18', [
+                            'type'    => Column::TYPE_BOOLEAN,
+                            'default' => 'true',
+                            'notNull' => true,
+                            'after'   => 'like',
+                            'comment' => 'test',
+                        ]),
+                    ],
+                    'options' => [
+                        "table_comment"=>"test TABLE_COMMENT"
```
Comment (Jeckerson) [inline tests/tocheck-database/Db/Dialect/Postgresql/CreateTableCest.php:256]:
```suggestion
                        'table_comment' => 'test TABLE_COMMENT',
```

**Review by Jeckerson — CHANGES_REQUESTED** [review]

File: phalcon/Db/Adapter/Pdo/Mysql.zep [hunk 16]
```diff
@@ -401,39 +401,44 @@ class Mysql extends PdoAdapter
...
+             }
```
Comment (Jeckerson) [inline phalcon/Db/Adapter/Pdo/Mysql.zep:441]:
```suggestion
             }

```

File: phalcon/Db/Adapter/Pdo/Postgresql.zep [hunk 17]
```diff
@@ -512,7 +512,12 @@ class Postgresql extends PdoAdapter
                     let definition["default"] = null;
                 }
             }
-
+            /**
```
Comment (Jeckerson) [inline phalcon/Db/Adapter/Pdo/Postgresql.zep:515]:
```suggestion

            /**
```

File: phalcon/Db/Adapter/Pdo/Postgresql.zep [hunk 18]
```diff
@@ -512,7 +512,12 @@ class Postgresql extends PdoAdapter
...
+             * Check if the column has comment
+             */
+             if field[10] !== null {
+                let definition["comment"] = "" . field[10];
+             }
```
Comment (Jeckerson) [inline phalcon/Db/Adapter/Pdo/Postgresql.zep:520]:
```suggestion
             }

```

File: phalcon/Db/Dialect/Postgresql.zep [hunk 19]
```diff
@@ -250,6 +259,14 @@ class Postgresql extends Dialect
         let sql .= join(",\n\t", createLines) . "\n)";
         if isset definition["options"] {
             let sql .= " " . this->getTableOptions(definition);
+            /**
```
Comment (Jeckerson) [inline phalcon/Db/Dialect/Postgresql.zep:262]:
```suggestion

            /**
```

File: phalcon/Db/Dialect/Postgresql.zep (hunk 11 above)
Comment (cq-z) [reply phalcon/Db/Dialect/Postgresql.zep:267]: set PGSQL table comment。
the sql is “comment on table table_name is 'xxx'; ”

Comment (cq-z) [reply phalcon/Db/Dialect/Postgresql.zep]: this is bug!

**Comment by ruudboon** [comment]
As I mentioned before, the`tests\tocheck-database` aren't part of the CI yet and need to be refactored to the format in `tests\database`. See for example https://github.com/phalcon/cphalcon/pull/15076

If you are putting in effort to the tests in tocheck-database it would be great if you can directly move those files to tests\tocheck-database and refactor them to use the new structure.

**Comment by cq-z** [comment]
@ruudboon Sorry, I don't quite understand what you mean! You mean move tests\tocheck-database change to tests\database?

**Comment by ruudboon** [comment]
> @ruudboon Sorry, I don't quite understand what you mean! You mean move tests\tocheck-database change to tests\database?

Yes, you can move them to there but they need to be adjusted to have the new structure. The pull I mentioned is a good example how to refactoring needs to be done. If you can refactor the tests you are now touching it would be great.

**Comment by cq-z** [comment]
@ruudboon CI has issue？ Do you need to reopen? or I am change？

**Comment by zsilbi** [comment]
@cq-z could you rebase this please? 
We would like to include this in 4.1

**Comment by cq-z** [comment]
ok，but have there an example ？

**Comment by zsilbi** [comment]
> ok，but have there an example ？

I am not sure what you mean, but you can do a rebase like this:
If your remote is called upstream:

```
git fetch upstream
git rebase upstream/4.1.x
```

**Comment by cq-z** [comment]
Sorry, misunderstanding. Is it OK now?

**Comment by zsilbi** [comment]
@cq-z there are still some requested changes left, could you please address them?

**Comment by cq-z** [comment]
@zsilbi Is it OK now?

**Review by Jeckerson — APPROVED** [review]

**Comment by zsilbi** [comment]
Thank you @cq-z

**Pull request merged**